connect/runtime/src/main/java/org/apache/kafka/connect/runtime/ConnectorStatus.java

```diff
@@ -0,0 +1,58 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ **/
+package org.apache.kafka.connect.runtime;
+
+public class ConnectorStatus extends AbstractStatus<String> {
+
+    public ConnectorStatus(String connector, State state, String msg, String workerUrl, int generation) {
+        super(connector, state, msg, workerUrl, generation);
+    }
+
+    public ConnectorStatus(String connector, State state, String workerUrl, int generation) {
+        super(connector, state, null, workerUrl, generation);
+    }
+
+    public interface Listener {
```

Comment:
That is true, although then I couldn't have AbstractHerder implement both interfaces. That's not a real limitation since I could create nested classes to implement them separately anyway... Mainly I duplicated the interface because ConnectorStatus.Listener reads easier than AbstractStatus.Listener<String>.